networkx/classes/digraph.py

```diff
@@ -898,23 +898,30 @@ def degree_iter(self, nbunch=None, weight=None):
         [(0, 1), (1, 2)]
 
         """
+        if nbunch in self:
+            if weight is None:
+                return len(self.succ[nbunch]) + len(self.pred[nbunch])
+            return sum(self.succ[nbunch][nbr].get(weight, 1) for nbr in self.succ[nbunch]) + \
+                   sum(self.pred[nbunch][nbr].get(weight, 1) for nbr in self.pred[nbunch])
+
```

Comment:
I would start with ```succ = self.succ[nbunch]``` and ```pred = self.pred[nbunch]```
Then the form of the code just below this should work. 

    return (nbunch ,len(succ)+len(pred))